### PR TITLE
feat/603: remove Auth0 related features

### LINKED ISSUE #306: SIT-3368: Use the new curator API in the plugin, which does not use Auth0 authentication
### Is your enhancement related to a problem? Please describe.

From Swati Matta:

> The new API will shorten the response time, and remove dependency on Auth0 authentication. Document attached. This will need only the following fields to be configured in the plugin:
> - Tenant ID
> - Host ID
> - On a widget level - page name and widget name (This is as-is behaviour from today)
>
> Please let me know if you have any questions after reviewing the documentation.

For historical context, see: https://voltron.atlassian.net/servicedesk/customer/portal/41/SIT-3368.

### Designs

n/a

### Describe alternatives you've considered

n/a

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

- Removed the Auth0 class and related code.
- Removed the client ID and secret fields.
- Added Host and Tenant ID fields.
- Added a cleanup function to delete client ID, client secret and auth token transient.

Closes #306 

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests passed.

### Changelog Entry

> Added - New Curator API.
> Removed - Dependency on Auth0 and replaced with the new Curator API.

### Credits

Props @Sidsector9

### CODE REVIEW

File: includes/functions/settings.php [hunk 1]
```diff
@@ -259,32 +256,6 @@ function sanitize_settings( $settings ) {
...
 	}
 
-	$prev_settings = get_option( SETTINGS_GROUP );
-
-	// Delete the option that holds the access token when the environment is changed
-	if ( ! empty( $prev_settings['environment'] ) && ! empty( $settings['environment'] ) && $prev_settings['environment'] !== $settings['environment'] ) {
-		delete_option( 'sophi_site_automation_access_token' );
-	}
-
-	if ( ! empty( $settings['client_id'] && ! empty( $settings['client_secret'] ) ) ) {
```
Comment (jeffpaul) [inline includes/functions/settings.php:269]: I think we may still want some validation that Host/Tenant ID are entered when someone tries to save the form.

**Review by dkotter — DISMISSED** [review]

File: sophi.php [hunk 2]
```diff
@@ -93,6 +93,49 @@ function() {
...
+		unset( $sophi_settings['client_id'] );
+		unset( $sophi_settings['client_secret'] );
+
+		update_option( 'sophi_settings', $sophi_settings );
+		update_option( $version_key, SOPHI_WP_VERSION, true );
+		set_transient( $version_key, SOPHI_WP_VERSION );
+	}
+}
+
+add_action( 'plugin_loaded', 'sophi_remove_stale_data' );
```
Comment (dinhtungdu) [inline sophi.php]: I'm a bit worried about the new option in terms of site performance. Is there another way to clean the data without adding a new option?

I think we can remove irrelevant options when updating the Sophi setting. Do you think it's sufficient to clean the old settings?

Comment (Sidsector9) [reply sophi.php]: @dinhtungdu I've replaced this logic, can you check?

File: sophi.php [hunk 3]
```diff
@@ -93,6 +93,34 @@ function() {
...
+		unset( $value['client_id'] );
+	}
+
+	if ( isset( $is_client_secret_set ) ) {
+		unset( $value['client_secret'] );
+	}
+
+	return $value;
+}
+add_filter( 'pre_update_option_sophi_settings', 'sophi_remove_stale_data' );
```
Comment (dinhtungdu) [inline sophi.php]: I think this logic can be added to the `sanitize_settings` function in `includes/functions/settings.php` instead.

**Review by dinhtungdu — COMMENTED** [review]
@Sidsector9 thanks for the update! I just have one more minor comment, otherwise, this is LGTM!

**Review by dinhtungdu — APPROVED** [review]

**Comment by dinhtungdu** [comment]
@Sidsector9 Thanks so much for being patient with me! This is LGTM now!

**Pull request merged**